src/m365/entra/commands/group/group-set.ts

```diff
@@ -0,0 +1,323 @@
+        }
+
+        if (args.options.newDisplayName && args.options.newDisplayName.length > 256) {
+          return `The maximum amount of characters for 'newDisplayName' is 256.`;
+        }
+
+        if (args.options.mailNickname) {
+          if (!validation.isValidMailNickname(args.options.mailNickname)) {
+            return `Value '${args.options.mailNickname}' for option 'mailNickname' must contain only characters in the ASCII character set 0-127 except the following: @ () \ [] " ; : <> , SPACE.`;
+          }
+
+          if (args.options.mailNickname.length > 64) {
+            return `The maximum amount of characters for 'mailNickname' is 64.`;
+          }
+        }
+
+        if (args.options.ownerIds) {
+          const isValidGUIDArrayResult = validation.isValidGuidArray(args.options.ownerIds);
+          if (isValidGUIDArrayResult !== true) {
+            return `The following GUIDs are invalid for the option 'ownerIds': ${isValidGUIDArrayResult}.`;
+          }
+        }
+
+        if (args.options.ownerUserNames) {
+          const isValidUPNArrayResult = validation.isValidUserPrincipalNameArray(args.options.ownerUserNames);
+          if (isValidUPNArrayResult !== true) {
+            return `The following user principal names are invalid for the option 'ownerUserNames': ${isValidUPNArrayResult}.`;
+          }
+        }
+
+        if (args.options.memberIds) {
+          const isValidGUIDArrayResult = validation.isValidGuidArray(args.options.memberIds);
+          if (isValidGUIDArrayResult !== true) {
+            return `The following GUIDs are invalid for the option 'memberIds': ${isValidGUIDArrayResult}.`;
+          }
+        }
+
+        if (args.options.memberUserNames) {
+          const isValidUPNArrayResult = validation.isValidUserPrincipalNameArray(args.options.memberUserNames);
+          if (isValidUPNArrayResult !== true) {
+            return `The following user principal names are invalid for the option 'memberUserNames': ${isValidUPNArrayResult}.`;
+          }
+        }
+
+        if (args.options.visibility && this.allowedVisibility.indexOf(args.options.visibility) === -1) {
+          return `Option 'visibility' must be one of the following values: Public, Private.`;
+        }
+
+        if (args.options.newDisplayName === undefined && args.options.description === undefined && args.options.visibility === undefined
+          && args.options.ownerIds === undefined && args.options.ownerUserNames === undefined && args.options.memberIds === undefined
+          && args.options.memberUserNames === undefined) {
+          return `Specify at least one of the following options: 'newDisplayName', 'description', 'visibility', 'ownerIds', 'ownerUserNames', 'memberIds', 'memberUserNames'`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      { options: ['id', 'displayName'] }
+    );
```

Comment:
Let's add optional option sets for options used to specify members and owners.

Reply:
ok